pkg/build/raw_test.go

```diff
@@ -43,60 +36,91 @@ func TestCreateRawImageCopyCommand(t *testing.T) {
+	defer teardown()
+	ctx.ImageDefinition = &image.Definition{
+		Image: image.Image{
+			OutputImageName: "output-image",
+		},
+		OperatingSystem: image.OperatingSystem{
+			KernelArgs: []string{"alpha", "beta"},
+		},
+	}
+	builder := Builder{context: ctx}
+	outputImageFilename := builder.generateOutputImageFilename()
+
+	tests := []struct {
+		name              string
+		includeCombustion bool
+		renameFilesystem  bool
+		expectedContains  []string
+		expectedMissing   []string
+	}{
+		{
+			name:              "RAW Image Usage",
+			includeCombustion: true,
+			renameFilesystem:  true,
+			expectedContains: []string{
+				fmt.Sprintf("guestfish --format=raw --rw -a %s", outputImageFilename),
+				fmt.Sprintf("copy-in %s", builder.context.CombustionDir),
+				"download /boot/grub2/grub.cfg /tmp/grub.cfg",
+				"btrfs filesystem label / INSTALL",
+			},
+			expectedMissing: []string{},
+		},
+		{
+			name:              "ISO Image Usage",
+			includeCombustion: false,
+			renameFilesystem:  false,
+			expectedContains: []string{
+				fmt.Sprintf("guestfish --format=raw --rw -a %s", outputImageFilename),
+				"download /boot/grub2/grub.cfg /tmp/grub.cfg",
+			},
+			expectedMissing: []string{
+				fmt.Sprintf("copy-in %s", builder.context.CombustionDir),
+				"btrfs filesystem label / INSTALL",
 			},
 		},
 	}
 
 	// Test
-	err = builder.writeModifyScript()
-
-	// Verify
-	require.NoError(t, err)
-
-	expectedFilename := filepath.Join(tmpDir, modifyScriptName)
-	foundBytes, err := os.ReadFile(expectedFilename)
-	require.NoError(t, err)
-
-	stats, err := os.Stat(expectedFilename)
-	require.NoError(t, err)
-	assert.Equal(t, fileio.ExecutablePerms, stats.Mode())
-
-	foundContents := string(foundBytes)
-	assert.Contains(t, foundContents, "guestfish --format=raw --rw -a config-dir/output-image")
-	assert.Contains(t, foundContents, "copy-in "+builder.context.CombustionDir)
-	assert.Contains(t, foundContents, "download /boot/grub2/grub.cfg /tmp/grub.cfg")
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := builder.writeModifyScript(outputImageFilename, test.includeCombustion, test.renameFilesystem)
+			require.NoError(t, err)
+
+			expectedFilename := filepath.Join(ctx.BuildDir, modifyScriptName)
+			foundBytes, err := os.ReadFile(expectedFilename)
+			require.NoError(t, err)
+
+			stats, err := os.Stat(expectedFilename)
+			require.NoError(t, err)
+			assert.Equal(t, fileio.ExecutablePerms, stats.Mode())
+			foundContents := string(foundBytes)
+
+			for _, findMe := range test.expectedContains {
+				assert.Contains(t, foundContents, findMe)
+			}
+			for _, dontFindMe := range test.expectedMissing {
+				assert.NotContains(t, foundContents, dontFindMe)
+			}
+		})
+	}
 }
 
 func TestCreateModifyCommand(t *testing.T) {
 	// Setup
-	builder := Builder{
-		context: &image.Context{
-			BuildDir: "build-dir",
-		},
-	}
+	ctx, teardown := setupContext(t)
+	defer teardown()
+	builder := Builder{context: ctx}
```

Comment:
This was intentionally not creating a fully populated context since it is not needed and not affecting the scenario we wanted to test. Is there a reason for changing it?

Reply:
Nope, was just changing for consistency. I can revert if you'd like.